patterns/observers/CinemaTicketBookingExample

```diff
@@ -0,0 +1,90 @@
+import java.util.ArrayList;
+import java.util.List;
+
+// Defines the Subject interface that allows observers to register, unregister, and be notified.
+interface TicketSubject {
+    void registerObserver(TicketObserver observer);
+    void removeObserver(TicketObserver observer);
+    void notifyObservers();
+}
+
+// Implements the TicketSubject interface to manage ticket availability and observers.
+class TicketBookingSystem implements TicketSubject {
+    private List<TicketObserver> observers; // List to hold all registered observers.
+    private boolean ticketAvailable; // Flag to track ticket availability status.
+
+    public TicketBookingSystem() {
+        observers = new ArrayList<>(); // Initialize the list of observers.
+        ticketAvailable = false; // Initially, tickets are not available.
+    }
+
+    // Sets the ticket availability and notifies all observers about the change.
+    public void setTicketAvailable(boolean available) {
```

Comment:
`setTicketAvailability` is probably clearer. `setTicketAvailable` sounds like the intention is to set the `ticketAvailable` field to `true` no matter what (i.e. 'available' vs 'unavailable' vs 'availability').